ci/scripts/run-in-nix-docker.sh

```diff
@@ -40,6 +43,7 @@ IMAGE_SHA=$(docker build --quiet -f ./ci/docker/nix.Dockerfile .)
 # build artifacts.
 # The nix image in use doesn't work if run as a non-root user.
 docker run --rm -it \
+	--env-file env-file-for-buildkite \
```

Comment:
huh.... Would there be any reason to not pass all environment variables in this way?

Reply:
I choose this approach as I didn't know how many buildkite env vars I need. I could remove fliter by grep command to pass all env vars to the container. What do you think?

Reply:
Hmmm, off the top of my head I'm not sure what the behavior will be if there are conflicts (thinking of `DOCKER_HOST`). Let's leave it as is so we don't have to keep spending more time here.